src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -617,18 +620,39 @@ int java_lang_String::utf8_length(oop java_string, typeArrayOop value) {
   }
 }
 
-int java_lang_String::utf8_length(oop java_string) {
+size_t java_lang_String::utf8_length(oop java_string) {
   typeArrayOop value = java_lang_String::value(java_string);
   return utf8_length(java_string, value);
 }
 
+int java_lang_String::utf8_length_as_int(oop java_string) {
+  typeArrayOop value = java_lang_String::value(java_string);
+  return utf8_length_as_int(java_string, value);
+}
+
+int java_lang_String::utf8_length_as_int(oop java_string, typeArrayOop value) {
```

Comment:
Why not call java_lang_String::utf8_length() here instead of duplicating the code?

Reply:
Because we have to call `UNICODE::utf8_length_as_int` to get the proper truncation - the code is not an exact duplicate.

Reply:
OK, I missed that.  It's probably not worth it to refactor out the common code using C++ lambda expressions.